File: hash/완주하지못한선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> cm, pm;
    for(auto c : completion) {
        cm[c]++;
    }
    
    for(auto p : participant) {
        pm[p]++;
    }
    
    for(auto p : pm) {
        if(cm.find(p.first) == cm.end() || cm[p.first] < p.second)
            return p.first;
    }
    
    return answer;
}

File: Hash/완주하지못한선수.java
import java.util.HashMap;
import java.util.Map;

class 완주하지못한선수 {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hm = new HashMap<>();
        for(String p: participant){
                hm.put(p, hm.getOrDefault(p, 0)+1);
        }
        
        for(String c: completion){
            hm.put(c, hm.get(c)-1);
        }
        
        for(Map.Entry<String, Integer> el: hm.entrySet()){
            if(el.getValue() > 0){
                answer = el.getKey();
                break;
            }
        }
        
        return answer;
    }
}